### ISSUE TITLE
Calling webSocket.isDataPending gives "operation was attempted on something that is not a socket"

### ISSUE THREAD

**Issue by CoderNate** [adamdruppe/arsd#354]
OS: Windows

The `readSet` here is static and in similar places where a `SocketSet` is used there is a call to `readSet.reset()` prior to calling `readSet.add(...)`.
https://github.com/adamdruppe/arsd/blob/f98b5e17d651ee815fd4ce787dac1541f1959f2d/http2.d#L4655-L4658

I'm seeing that `isDataPending` works okay for the first WebSocket I create in a session but after I close it and create a new WebSocket one of two things happen:
* Calling `isDataPending` with a timeout just keeps returning false immediately every time it's called
* I get `std.socket.SocketOSException@std\socket.d(3488): Socket select error: An operation was attempted on something that is not a socket.`

If I add in a call to `readSet.reset()` then everything seems to work fine.

**Comment by adamdruppe** [comment]
that's right i am supposed to reset it, i just barely ever use this function so i didn't test it well. pushed fix up.
